cmd/tikv-ctl/src/util.rs

```diff
@@ -62,8 +63,39 @@ pub fn perror_and_exit<E: Error>(prefix: &str, e: E) -> ! {
     tikv_util::logger::exit_process_gracefully(-1);
 }
 
+// Check if the part of region's `key_range` is in the specified
+// `key_range_limit`.
+pub fn check_range_included(key_range: &KeyRange, key_range_limit: &KeyRange) -> bool {
+    // When limit start_key contains region start_key
+    if key_range_limit.get_start_key() <= key_range.get_start_key()
+        && (key_range_limit.get_end_key().is_empty() /* when limit end_key is infinite */
+         || key_range_limit.get_end_key() >= key_range.get_start_key())
+    {
+        return true;
+    }
+    // When limit end_key contains region end_key
+    if (key_range_limit.get_end_key() >= key_range.get_end_key()
+        || key_range_limit.get_end_key().is_empty())
+        && key_range_limit.get_start_key() < key_range.get_end_key()
+    {
+        return true;
+    }
+    // When `key_range_limit` is in the region's `key_range` also need to
+    // be considered as included.
+    if key_range.get_start_key() <= key_range_limit.get_start_key()
+        && (key_range.get_end_key().is_empty() /* when region end_key is infinite */
+            || (!key_range_limit.get_end_key().is_empty() /* when limit end_key is not infinite */
+                    && key_range.get_end_key() >= key_range_limit.get_end_key()))
+    {
+        return true;
+    }
+    false
+}
```

Comment:
```rust
if !a.get_end_key().is_empty() && !b.get_start_key().is_empty() && a.get_end_key() < b.get_start_key() {
  false
}
if !b.get_end_key().is_empty() && !a.get_start_key().is_empty() && b.get_end_key() < a.get_start_key() {
  false
}
true
```

Reply:
Thx for your help, the not contained case is indeed simpler than the contained case.
BTW, since it is a left-closed right-open interval, I added the = judgment:)
Plz take a look, thx!!! @tabokie @JmPotato